### PR TITLE
Vehicle testing

### LINKED ISSUE #49: Mock Data - Add Mock Data for Everything
### Description

We need to mock a ton of data so we can reliably test without going on the car. we need a mock current data with at least 40 elements. and to mock every item that we display on the screen. 

Also we want to occasionally send None items so that we can reliably test and type check functions that we use. 

### Acceptance Criteria

No MORE N/As 

### Proposed Solution

add mock data to the mock model in the style that weve been doing. 

### Mocks

_No response_

### LINKED ISSUE #51: Miscellaneous Improvements
### Description

there are some weird UI things on the car, 100% are too big on some components, we should add padding between the value column on the debug table and some other stuff. 

### Acceptance Criteria

generally looks good

### Proposed Solution

bruh

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by Peyton-McKee**
## Changes

A ton of changes making a PR cause we should prob merge in these changes now. Theyre mostly bug fixes and stuff. All my optimization attempts failed miserably which is why theres a ton of reverts. 

## Notes

Added Mock data here too cause i was bored

## Screenshots
Screen Shot 2023-04-15 at 4 14 17 AM

Screen Shot 2023-04-15 at 4 27 07 AM

Screen Shot 2023-04-15 at 4 14 43 AM

Screen Shot 2023-04-15 at 4 15 03 AM

Screen Shot 2023-04-15 at 4 15 25 AM

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes #49 
Closes #51

### CODE REVIEW

File: modes/debug_mode/debug_utils.py [hunk 1]
```diff
@@ -15,7 +15,7 @@ def __init__(self, fault_decimal: int, max_cell_temp: int, max_cell_voltage: int
         fault_bin = bin(fault_decimal)[2:]
         for i in range(0, len(fault_bin)):
             if fault_bin[i] == '1':
-                self.faults.append(FAULTS[i])
+                self.faults.append(FAULTS[i + 1])
```
Comment (Peyton-McKee) [inline modes/debug_mode/debug_utils.py]: this is awaiting whether or not this is correct

**Review by Peyton-McKee — COMMENTED** [review]
Also we may be buying a new Pi with 8 gigs of ram rather than the 1 gig that were using, if we dont, or that doesnt end up improving performance, we may have to efficiency and speed to not use matplotlibs because those eat up CPU / memory

File: nero_view.py [hunk 2]
```diff
@@ -89,7 +89,7 @@ def check_can(self):
 
     def update_current_page(self):
         self.current_screen.current_page.update()
-        self.after(100, self.update_current_page)
+        self.after(250, self.update_current_page)
```
Comment (Peyton-McKee) [inline nero_view.py:92]: I want to make this 50 ideally aka 20 fps, buttttttt I cant get it to work without tanking the pi, even without graphs

File: models/mock_model.py [hunk 3]
```diff
@@ -18,22 +19,52 @@ def __init__(self) -> None:
...
+        self.segement3_temp = 35
+        self.segement4_temp = 15
+        self.motor_power = 100
+        self.fan_power = 100
+        self.torque_power = 100
+        self.regen_power = 1
+        self.mode_index = 0
+        self.state_of_charge_delta = -1
+        self.inverter_temp = 30
         self.current_data = [self.mph, self.status, self.dir, self.pack_temp,
```
Comment (anthonybernardi) [inline models/mock_model.py:59]: get rid of this and just make the get functions return teh actual values

it's a mock model, it's just supposed to output the right stuff the logic doesnt need to be similar at all

File: components/circular_progress.py [hunk 4]
```diff
@@ -38,6 +38,29 @@ def set(self, value):
             color = "yellow"
         elif self.extent > 20:
             color = "orange"
+        if self.extent >= 100:
+            self.itemconfigure(self.label_id, font=tkFont.Font(family="Lato", size=int(((self.x1 - self.x0) / 3) - 10), weight='bold'))
+        self.itemconfigure(self.oval_id2, outline=color)
+        self.itemconfigure(self.arc_id, outline=color)
+        self.itemconfigure(self.oval_id1, outline=color)
+
+    def setSettingValue(self, value, numSettings=3):
```
Comment (anthonybernardi) [inline components/circular_progress.py]: this is a terrible function name, plz make it better

(and also add python docstring https://peps.python.org/pep-0257/)

File: nero_view.py (hunk 2 above)
Comment (anthonybernardi) [reply nero_view.py:92]: rip

**Review by anthonybernardi — CHANGES_REQUESTED** [review]
pretty nice

File: models/mock_model.py (hunk 3 above)
Comment (Peyton-McKee) [reply models/mock_model.py:59]: lets have new ppl do this

**Review by anthonybernardi — APPROVED** [review]

**Pull request merged**